Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpUnitTestsRunnerExecutionStrategy.cs

```diff
@@ -0,0 +1,226 @@
+                var arguments = new List<string> { dllPath };
+                arguments.AddRange(executionContext.AdditionalCompilerArguments.Split(' '));
+
+                // Run unit tests on the resulting .dll
+                var processExecutionResult = executor.Execute(
+                    this.NUnitConsoleRunnerPath,
+                    string.Empty,
+                    executionContext.TimeLimit,
+                    executionContext.MemoryLimit,
+                    arguments);
+
+                // Construct and figure out what the Test result is
+                TestResult testResult = new TestResult()
+                {
+                    Id = test.Id,
+                    TimeUsed = (int)processExecutionResult.TimeWorked.TotalMilliseconds,
+                    MemoryUsed = (int)processExecutionResult.MemoryUsed
+                };
+
+                switch (processExecutionResult.Type)
+                {
+                    case ProcessExecutionResultType.RunTimeError:
+                        testResult.ResultType = TestRunResultType.RunTimeError;
+                        testResult.ExecutionComment = processExecutionResult.ErrorOutput.MaxLength(2048); // Trimming long error texts
+                        break;
+                    case ProcessExecutionResultType.TimeLimit:
+                        testResult.ResultType = TestRunResultType.TimeLimit;
+                        break;
+                    case ProcessExecutionResultType.MemoryLimit:
+                        testResult.ResultType = TestRunResultType.MemoryLimit;
+                        break;
+                    case ProcessExecutionResultType.Success:
+                        int totalTests = 0;
+                        int passedTests = 0;
+                 
+                        this.ExtractTestResult(processExecutionResult.ReceivedOutput, ref passedTests, ref totalTests);
+                        string message = "Test Passed!";
+
+                        if (totalTests == 0)
+                        {
+                            message = "No tests found";
+                        }
+                        else if (passedTests == originalTestsPassed)
+                        {
+                            message = "No functionality covering this test!";
+                        }
+
+                        if (count == 0)
+                        {
+                            originalTestsPassed = passedTests;
+                            if (totalTests != passedTests)
+                            {
+                                message = "Not all tests passed on the correct solution.";
+                            }
+                        }
+
+                        var checkerResult = checker.Check(test.Input, message, test.Output, test.IsTrialTest);
+                        testResult.ResultType = checkerResult.IsCorrect ? TestRunResultType.CorrectAnswer : TestRunResultType.WrongAnswer;
```

Comment:
Place `? TestRunResultType.CorrectAnswer` on new line and `: TestRunResultType.WrongAnswer;` right below.